cloud-openstack-agent/src/main/java/jetbrains/buildServer/clouds/openstack/OpenstackAgentPropertiesReader.java

```diff
@@ -41,6 +42,23 @@ public void afterAgentConfigurationLoaded(@NotNull BuildAgent agent) {
 
             JsonElement metadataElement = new JsonParser().parse(rawMetadata);
 
+            JsonElement teamCityUserData = metadataElement.getAsJsonObject().get("meta");
+            Type type = new TypeToken<HashMap<String, String>>() {}.getType();
+            HashMap<String,String> customParameters = new Gson().fromJson(teamCityUserData, type);
+
+            if (!customParameters.containsKey(OpenstackCloudParameters.SERVER_URL)) {
+                LOG.info("Not my OpenStack instance, skipping");
```

Comment:
What is the goal to configure URL & name of TeamCity server ? This informations are provided by cloud process. If it is to use an Openstack agent as classic agent, is it duplicate of  #23 ?